doc/west/repo-tool.rst

```diff
@@ -0,0 +1,266 @@
+     (see below).
+
+- ``west pull [PROJECT ...]``: Fetches new upstream changes in each of the
+  specified projects (default: all projects) and rebases them to
+  ``manifest-rev``.
+
+  This corresponds to ``git pull --rebase``, with the upstream branch taken as
+  ``manifest-rev``.
+
+- ``west rebase [PROJECT ...]``: Rebases each of the specified projects
+  (default: all projects) to ``manifest-rev``.
+
+  .. note::
+
+     ``west pull`` is an alias for ``west fetch`` + ``west rebase``.
+
+Working with branches
+*********************
+
+The following commands are used to create, check out, and list branches that
+span multiple projects (in reality, Git branches with identical names in
+multiple repositories).
+
+- ``west branch [BRANCH_NAME [PROJECT ...]]``: Creates a branch
+  ``BRANCH_NAME`` in each of the specified projects (default: all projects).
+
+  Like all branches created by the multi-repo commands, the newly created
+  branches track ``manifest-rev``.
+
+  If no arguments are passed to ``west branch``, local branches from all
+  projects are listed, along with the projects they appear in.
+
+- ``west checkout [-b] BRANCH_NAME [PROJECT ...]]``: Checks out the branch
+  ``BRANCH_NAME`` in each of the specified projects (default: all projects).
+  This command is a no-op for projects that do not have the specified branch.
+
+  With ``-b``, creates and checks out the branch if it does not exist in a
+  project (like ``git checkout -b``), instead of ignoring the project.
+
+Miscellanous commands
+*********************
+
+These commands perform miscellaneous functions.
+
+- ``west list``: Lists project information from the manifest (URL, revision,
+  path, etc.), along with other manifest-related information.
+
+- ``west diff [PROJECT ...] [ARGUMENT ...]``: Runs a multi-repo ``git diff``
+  for the specified projects (default: all projects).
+
+  Extra arguments are passed as-is to ``git diff``.
+
+- ``west status [PROJECT ...] [ARGUMENT ...]``: Like ``west diff``, for
+  running ``git status``.
+
+- ``west update [--update-west] [--update-manifest]``: Updates the west and
+  manifest repositories.
+
+  Normally, the west and manifest repositories are updated automatically
+  whenever a command that fetches upstream data is run. This self-update is
+  skipped (with a warning) if not possible via fast-forward (i.e., if you've
+  made modifications to the west or manifest repositories).
+
+Example workflow
+================
+
+This section gives an example workflow that clones the Zephyr repositories,
+creates and switches between two work branches, updates (rebases) a work branch
+to the latest manifest revision, and inspects upstream changes.
+
+1. First, we initialize West with the default Zephyr manifest, and fetch all
+   projects:
+
+   .. code-block:: console
+
+      $ west init
+      $ west fetch
+
+   .. note::
+
+      ``west clone`` could be used instead of ``west fetch``, to automatically
```

Comment:
Was thinking of having `west clone` first, but thought this would be a nice and friendly way to explain detached HEADs and the difference between `fetch` and `clone`, which will probably come up elsewhere.

Any other opinions? Could change it if you think it'd be super confusing.

Reply:
I do believe that we should have both, but the default should use `west clone` because most people will use this as a reference. So I would reverse the main content and the note.

Reply:
Alright, will go for that.